tokio/src/executor/task/raw.rs

```diff
@@ -9,13 +9,13 @@ use std::ptr::NonNull;
 use std::task::Waker;
 
 /// Raw task handle
-pub(super) struct RawTask<S: 'static> {
-    ptr: NonNull<Header<S>>,
+pub(super) struct RawTask {
+    ptr: NonNull<Header>,
```

Comment:
Do we want to introduce a `PhantomData<S>` here as well? Would prevent a lot of the unsafety below.